src/item.h

```diff
@@ -206,6 +207,226 @@ class ItemAttributes
 			return static_cast<ItemDecayState_t>(getIntAttr(ITEM_ATTRIBUTE_DECAYSTATE));
 		}
 
+		struct CustomAttributeKey
+		{
+			union {
+				std::string* string;
+				int64_t integer;
+			} value;
```

Comment:
This union should be a separate class (or preferably use boost::variant/boost::any instead).